diff --git "a/C:\\Users\\morjos\\AppData\\Local\\Temp\\TortoiseGit\\DataReaderImpl-5c0fa3c.002.h" "b/E:\\OpenDDS\\dds\\DCPS\\DataReaderImpl.h"
index 616dc1816..2ab3faeca 100644
--- "a/C:\\Users\\morjos\\AppData\\Local\\Temp\\TortoiseGit\\DataReaderImpl-5c0fa3c.002.h"
+++ "b/E:\\OpenDDS\\dds\\DCPS\\DataReaderImpl.h"
@@ -727,11 +727,12 @@ private:
       , liveliness_timer_id_(-1)
     { }
 
-    void check_liveliness()
+	void check_liveliness();
+    /*void check_liveliness()
     {
       CheckLivelinessCommand c(this);
       execute_or_enqueue(c);
-    }
+    }*/
 
     void cancel_timer()
     {
