diff --git a/dds/DCPS/DataReaderImpl.h b/dds/DCPS/DataReaderImpl.h
index d7c5ede9f..606ad0a3b 100644
--- a/dds/DCPS/DataReaderImpl.h
+++ b/dds/DCPS/DataReaderImpl.h
@@ -27,6 +27,7 @@
 #include "Stats_T.h"
 #include "OwnershipManager.h"
 #include "ContentFilteredTopicImpl.h"
+#include "MultiTopicImpl.h"
 #include "GroupRakeData.h"
 #include "CoherentChangeControl.h"
 #include "AssociationData.h"
@@ -473,6 +474,12 @@ public:
 
   DDS::ContentFilteredTopic_ptr get_cf_topic() const;
 
+#endif
+
+#ifndef OPENDDS_NO_MULTI_TOPIC
+
+  void enable_multi_topic(MultiTopicImpl* mt);
+
 #endif
 
   void update_subscription_params(const DDS::StringSeq& params) const;
@@ -641,6 +648,9 @@ protected:
   TopicDescriptionPtr<ContentFilteredTopicImpl> content_filtered_topic_;
 #endif
 
+#ifndef OPENDDS_NO_MULTI_TOPIC
+  TopicDescriptionPtr<MultiTopicImpl> multi_topic_;
+#endif
 
   /// Is accessing to Group coherent changes ?
   bool coherent_;
@@ -753,11 +763,7 @@ private:
       , liveliness_timer_id_(-1)
     { }
 
-    void check_liveliness()
-    {
-      CheckLivelinessCommand c(this);
-      execute_or_enqueue(c);
-    }
+    void check_liveliness();
 
     void cancel_timer()
     {
