diff --git a/dds/DCPS/DataReaderImpl.h b/dds/DCPS/DataReaderImpl.h
index b301c0403..606ad0a3b 100644
--- a/dds/DCPS/DataReaderImpl.h
+++ b/dds/DCPS/DataReaderImpl.h
@@ -763,11 +763,7 @@ private:
       , liveliness_timer_id_(-1)
     { }
 
-    void check_liveliness()
-    {
-     CheckLivelinessCommand c(this);
-     execute_or_enqueue(c);
-    }
+	void check_liveliness();
 
     void cancel_timer()
     {
