Subject: [PATCH] Initialize structure enum fields
---
Index: TAO/TAO_IDL/be/be_visitor_field/field_ch.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/TAO/TAO_IDL/be/be_visitor_field/field_ch.cpp b/TAO/TAO_IDL/be/be_visitor_field/field_ch.cpp
--- a/TAO/TAO_IDL/be/be_visitor_field/field_ch.cpp	(revision f3832da16440b1eedbaba4dd4d324e8616cbcdbf)
+++ b/TAO/TAO_IDL/be/be_visitor_field/field_ch.cpp	(date 1685107058433)
@@ -56,7 +56,21 @@
     }
 
   // Now output the field name.
-  *os << " " << node->local_name () << ";";
+  *os << " " << node->local_name ();
+
+  bool is_scope_struct = false;
+  be_scope * scope = node->scope();
+  if (scope) {
+     is_scope_struct = scope->decl()->node_type() == AST_Decl::NT_struct;
+  }
+
+  if (node->field_type()->node_type() == AST_Decl::NT_enum && is_scope_struct) {
+    AST_Enum* enum_node = dynamic_cast<AST_Enum*>(node->field_type());
+    UTL_IdList* name = enum_node->value_to_name(0);
+    *os << " = " << node->field_type()->full_name() << "::" << name->last_component()->get_string();
+  }
+
+  *os << ";";
 
   return 0;
 }
