diff --git a/dds/DCPS/MultiTopicDataReaderBase.cpp b/dds/DCPS/MultiTopicDataReaderBase.cpp
index 73bc21908..0d44c5f50 100644
--- a/dds/DCPS/MultiTopicDataReaderBase.cpp
+++ b/dds/DCPS/MultiTopicDataReaderBase.cpp
@@ -108,6 +108,7 @@ void MultiTopicDataReaderBase::init(const DDS::DataReaderQos& dr_qos,
     return;
   }
 
+  resulting_impl->enable_multi_topic(multitopic);
   resulting_impl->raw_latency_buffer_size() = parent->raw_latency_buffer_size();
   resulting_impl->raw_latency_buffer_type() = parent->raw_latency_buffer_type();
 
