Index: dds/DCPS/DataWriterImpl.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/dds/DCPS/DataWriterImpl.cpp b/dds/DCPS/DataWriterImpl.cpp
--- a/dds/DCPS/DataWriterImpl.cpp	(revision 0d85140f59fcda8fa38064ca280e73e1b99702bf)
+++ b/dds/DCPS/DataWriterImpl.cpp	(revision 608c8f4d46107373bc66e7cbd4b638e4533b3440)
@@ -2521,6 +2521,9 @@
     ++liveliness_lost_status_.total_count;
     ++liveliness_lost_status_.total_count_change;
 
+    set_status_changed_flag(DDS::LIVELINESS_LOST_STATUS, true);
+    notify_status_condition();
+
     DDS::DataWriterListener_var listener = listener_for(DDS::LIVELINESS_LOST_STATUS);
 
     if (!CORBA::is_nil(listener.in())) {
